apps/openmw/mwworld/store.cpp

```diff
@@ -179,16 +185,19 @@ namespace MWWorld
         return ptr;
     }
     template<typename T>
-    void Store<T>::load(ESM::ESMReader &esm, const std::string &id) 
+    RecordId Store<T>::load(ESM::ESMReader &esm) 
     {
-        std::string idLower = Misc::StringUtils::lowerCase(id);
+        T record;
+        bool isDeleted = false;
 
-        std::pair<typename Static::iterator, bool> inserted = mStatic.insert(std::make_pair(idLower, T()));
+        record.load(esm, isDeleted);
+        Misc::StringUtils::toLower(record.mId);
+
+        std::pair<typename Static::iterator, bool> inserted = mStatic.insert(std::make_pair(record.mId, record));
         if (inserted.second)
             mShared.push_back(&inserted.first->second);
 
-        inserted.first->second.mId = idLower;
-        inserted.first->second.load(esm);
```

Comment:
That raises a problem... to merge records in this way, we need the ID of the record, <i>before</i> we load the rest of the record.

Question: do we need merging? It seems like the superior approach, but right now the ESM records save all fields anyway (even non-modified fields). But we might want to expand on this in the future. @zinnschlag: your opinion please?